simulator/func_sim/alu.h

```diff
@@ -332,6 +332,20 @@ struct ALU
         auto data = circ_ls( instr->v_src2, off);
         instr->v_dst = ( data & mask) | ( instr->v_src1 & ~mask);
     }
+  
+    // RISC-V Shuffle
+    template<typename I> static
+    void riscv_unshfl( I* instr)
+    {
+        auto pre_result = instr->v_src1;
+        size_t limit = 4 + bytewidth<decltype(instr->v_src1)> / bytewidth<uint64_t>;
```

Comment:
Do I understand correctly:
`uint32 -> 4`
`uint64 -> 5`
`uint128 -> 6`
?

Reply:
Yes, it is correct. In the specification there is formula `uint_xlen -> log2(XLEN) - 1`, but I think in this case 1 division is better than 1 logarithm.